watchmal/engine/regression.py

```diff
@@ -93,3 +93,36 @@ def restore_state(self, weight_file):
         super().restore_state(weight_file)
         if "target_sizes" in self.state_data:
             self.target_sizes = self.state_data["target_sizes"]
+
+class RegressionDIEngine(RegressionEngine):
+    """
+    Regression engine for dual-image input (20-inch image + mPMT image).
+    """
+
+    def __init__(self, target_key, model, rank, device, dump_path,
+                 target_scale_offset=0, target_scale_factor=1):
+        super().__init__(target_key, model, rank, device, dump_path,
+                         target_scale_offset=target_scale_offset,
+                         target_scale_factor=target_scale_factor)
+        self.data_main = None
+        self.data_second = None
+    def process_data(self, data):
+        data_main, data_second = data["data"]
+        self.data_main = data_main.to(self.device)
+        self.data_second = data_second.to(self.device)
+        self.data = (self.data_main, self.data_second)
```

Comment:
I think the existing process_data already does this [here](https://github.com/WatChMaL/WatChMaL/blob/924dd5403f3ea2c6fea744f869238304f15f81fb/watchmal/engine/reconstruction.py#L152-L157).